requestor/ext/requestor-gwtjackson/src/main/java/io/reinert/requestor/ext/gwtjackson/JsonObjectSerdesSchema.java

```diff
@@ -0,0 +1,339 @@
+        }
+    }
+
+    class LinkedHashSetReaderField extends FieldAssembler {
+        protected FieldSpec.Builder getDeclaration() {
+            return FieldSpec.builder(linkedHashSetReaderInterface.className(), "linkedHashSetReader")
+                    .addModifiers(Modifier.PRIVATE);
+        }
+    }
+
+    class TreeSetReaderField extends FieldAssembler {
+        protected FieldSpec.Builder getDeclaration() {
+            return FieldSpec.builder(treeSetReaderInterface.className(), "treeSetReader")
+                    .addModifiers(Modifier.PRIVATE);
+        }
+    }
+
+    class Constructor extends MethodAssembler {
+        protected MethodSpec.Builder getSignature() {
+            return MethodSpec.constructorBuilder()
+                    .addModifiers(Modifier.PUBLIC);
+        }
+    }
+
+    class GetLinkedListReaderMethod extends MethodAssembler {
+        protected MethodSpec.Builder getSignature() {
+            return MethodSpec.methodBuilder("getLinkedListReader")
+                    .returns(linkedListReaderInterface.className())
+                    .addModifiers(Modifier.PRIVATE);
+        }
+    }
+
+    class GetHashSetReaderMethod extends MethodAssembler {
+        protected MethodSpec.Builder getSignature() {
+            return MethodSpec.methodBuilder("getHashSetReader")
+                    .returns(hashSetReaderInterface.className())
+                    .addModifiers(Modifier.PRIVATE);
+        }
+    }
+
+    class GetLinkedHashSetReaderMethod extends MethodAssembler {
+        protected MethodSpec.Builder getSignature() {
+            return MethodSpec.methodBuilder("getLinkedHashSetReader")
+                    .returns(linkedHashSetReaderInterface.className())
+                    .addModifiers(Modifier.PRIVATE);
+        }
+    }
+
+    class GetTreeSetReaderMethod extends MethodAssembler {
+        protected MethodSpec.Builder getSignature() {
+            return MethodSpec.methodBuilder("getTreeSetReader")
+                    .returns(treeSetReaderInterface.className())
+                    .addModifiers(Modifier.PRIVATE);
+        }
+    }
+
+    class ReadJsonMethod extends MethodAssembler {
+        final ParameterSpec reader = ParameterSpec.builder(JsonRecordReader.class, "r").build();
+        final ParameterSpec context = ParameterSpec.builder(DeserializationContext.class, "ctx").build();
+
+        protected MethodSpec.Builder getSignature() {
+            // #readJson - used when none of deserialize alternatives succeeded (see JsonObjectSerdes)
+            return MethodSpec.methodBuilder(JsonObjectSerdesMeta.Method.READ_JSON)
+                    .returns(typeInfo.getClassName())
+                    .addParameter(reader)
+                    .addParameter(context)
+                    .addModifiers(Modifier.PUBLIC)
+                    .addAnnotation(Override.class);
+        }
+    }
+
+    class WriteJsonMethod extends MethodAssembler {
+        // #writeJson - not used
```

Comment:
Not used in the generated class, but necessary to be a valid class (this is an abstract method in the parent class)